src/components/structures/MatrixChat.js

```diff
@@ -1612,20 +1631,7 @@ export default React.createClass({
             dis.dispatch(payload);
         } else if (screen.indexOf('user/') == 0) {
             const userId = screen.substring(5);
-
-            // Wait for the first sync so that `getRoom` gives us a room object if it's
-            // in the sync response
-            const waitFor = this.firstSyncPromise ?
-                this.firstSyncPromise.promise : Promise.resolve();
-            waitFor.then(() => {
-                if (params.action === 'chat') {
-                    this._chatCreateOrReuse(userId);
-                    return;
-                }
-                this.notifyNewScreen('user/' + userId);
-                this.setState({currentUserId: userId});
-                this._setPage(PageTypes.UserView);
-            });
+            this._viewUser(userId, params.action);
```

Comment:
Should we change to dispatch `view_user_info` here instead of calling directly? It looks like many of the other branches dispatch actions, so just wondering it would make sense to follow that pattern.